lib/kubernetes-deploy/resource_cache.rb

```diff
@@ -36,6 +36,12 @@ def get_all(kind, selector = nil)
       []
     end
 
+    def prewarm(resources)
+      sync_dependencies = resources.map(&:class).flat_map { |c| c.const_get(:SYNC_DEPENDENCIES) }
```

Comment:
Let's avoid the double map:
`resources.flat_map { |r| r.class.const_get(:SYNC_DEPENDENCIES) }`